### DISCUSSION TITLE
@capacitor-firebase/firestore adding 1000s lines of pods

### DISCUSSION THREAD

**Discussion by jongbonga** [capawesome-team/capacitor-firebase#445] [General]
After replacing `@angular/fire` with `@capacitor-firebase/firestore`, build and sync my app, I noticed more than 1k lines of pod added.
notably `abseil`, `BoringSSL-GRPC`, `gRPC-Core`, `leveldb-library` & `Libuv-gRPC`.  

<img width="459" alt="Screenshot 2023-09-29 at 20 48 52" src="https://github.com/capawesome-team/capacitor-firebase/assets/12111716/fa2eaca3-b179-4c89-ade0-06545f2bda6f">

I understand the plugin is still on dev according to its versioning. What I want to know is if those PODs are required for the plugin to work and what would be the implication in terms of bundle size

**Comment by robingenz** [comment]
Yes, those pods are required. The only dependency that the plugin needs (excluding Capacitor) is `FirebaseFirestore` (see https://github.com/capawesome-team/capacitor-firebase/blob/1338d5fc38b4a91be118be8874336f75373093f7/packages/firestore/CapacitorFirebaseFirestore.podspec#L16). These are all sub-dependencies of `FirebaseFirestore`. I haven't looked at the bundle size yet. But even if the bundle size is big, the [advantages](https://twitter.com/robin_genz/status/1707647013897023697) outweigh the disadvantages for me.

**Reply by jongbonga** [reply]
Thanks for the quick reply. 

I was surprised since your other Firebase plugins come with minimal to no sub-dependencies.

**Reply by jongbonga** [reply]
My app the size doubled, I have an extra 70mb or so.

**Reply by robingenz** [reply]
Thanks for sharing! This SO answer contains some more information: https://stackoverflow.com/a/48363445/6731412
